lib/web_ui/lib/src/engine/text/line_breaker.dart

```diff
@@ -466,6 +493,24 @@ LineBreakResult nextLineBreak(String text, int index) {
       continue;
     }
 
+    // Break between two regional indicator symbols if and only if there are an
+    // even number of regional indicators preceding the position of the break.
+    // LB30a: sot (RI RI)* RI × RI
+    //        [^RI] (RI RI)* RI × RI
+    if (curr == LineCharProperty.RI) {
+      if (regionalIndicatorCount.isOdd) {
+        continue;
+      } else {
+        return LineBreakResult(index, LineBreakType.opportunity);
+      }
+    }
+
+    // Do not break between an emoji base and an emoji modifier.
+    // LB30b: EB × EM
+    if (prev1 == LineCharProperty.EB && curr == LineCharProperty.EM) {
```

Comment:
The most important line of code in this entire file 😉

Reply:
lol